packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,39 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+export function getVarSize(value: any): number {

```

Comment:
Possible to type this better than just `any`?

Reply:
I "learned" typescript 2 weeks ago, so if you have any suggestions for better typing please let me know. My IDE suggested `never`, but that came with other issues. Not sure if I should restrict it to some union of:
* hexstring
* number
* array of hex strings
* array serializable objects (specifically this one will be tricky as there is no interface afaik)

Reply:
Unions would be a good start. In typescript, the types are merely a signal so a serializable interface is match as long as the underneath type has the correct method signatures.
```
interface NeonSerializable {
  serialize(): string;
}
export function getVarSize(
  value: number | string | NeonSerializable | NeonSerializable[]
): number {
  return 1;
}

getVarSize(new Witness());
getVarSize(new Transaction());
getVarSize(1);
getVarSize("hi");
```

Reply:
shall we add `size(): number` to the `NeonSerializable` interface? 

Of course we can work around it by calling `serialize() / 2`, but I think that classes that are serializable should in general support `size()`.

Reply:
Is there a smarter way of doing?
```typescript

      } else if (
        typeof value.size === "number" &&
        typeof value.serialize === "function") {
        return getVarSize(value.size) + value.size;
```
because `value instanceof NeonSerializable` doesn't work, and the remaining code complains that `NeonSerializable` doesn't have a property `serialize` 😕

Reply:
Lets leave it there for now. I think a better way would be to formally introduce a base class for this.

Typescript transpiles to Javascript so all the interface magic goes away. Its all a facade for type checking so you cant actually attempt to apply javascript logic to types.